### ISSUE TITLE
[UX] Adjustments and  Updates to  widgets on Questionnaire  - SDC

### ISSUE THREAD

**Issue by f-odhiambo** [opensrp/fhircore#1974]
**Name of feature to enhance**

**Description of feature**

**Describe the enhancement**

**Comment by FikriMilano** [comment]
The SDK allows us to override the QuestionnaireItem widget's style or dimension.

The catch is that dimens may be used not only by the component we want to change but also by other components.

e.g. say we want to change the margin top for widgets, in some parts of the SDK, padding_default is used as a top margin for widgets, but it is also used for check box view padding; if we override that dimens property, it will affect both, which we don't want to happen. 

NOTE: I don't know why they use padding_default as margin top, it feels wrong.

For overriding styles, here's one example of the requirement we can implement on FHIR Core.
```
the questions font weights and size is different from designs. We use big font in these forms for
```

Here's one example of the requirement we can't implement on FHIR Core, need to do this in SDK.
```
For required questions, we shouldn’t use “*”, instead if a question is optional that’s when we will indicate the “(optional)” badge/text
```

**Comment by FikriMilano** [comment]
For the `*` or `optional` there is currently an open discussion here w @shelaghm
https://github.com/google/android-fhir/issues/544#issuecomment-1401237554

**Comment by dubdabasoduba** [comment]
> The SDK allows us to override the QuestionnaireItem widget's style or dimension.
> 
> The catch is that dimens may be used not only by the component we want to change but also by other components.
> 
> e.g. say we want to change the margin top for widgets, in some parts of the SDK, padding_default is used as a top margin for widgets, but it is also used for check box view padding; if we override that dimens property, it will affect both, which we don't want to happen.
> 
> NOTE: I don't know why they use padding_default as margin top, it feels wrong.
> 
> For overriding styles, here's one example of the requirement we can implement on FHIR Core.
> 
> ```
> the questions font weights and size is different from designs. We use big font in these forms for
> ```
> 
> Here's one example of the requirement we can't implement on FHIR Core, need to do this in SDK.
> 
> ```
> For required questions, we shouldn’t use “*”, instead if a question is optional that’s when we will indicate the “(optional)” badge/text
> ```
@FikriMilano are we overriding this by updating the `styles.xml` on FHIR Core?  Just making sure we will not be adding anything that might increase the complexity of keeping up with the SDK.

**Comment by FikriMilano** [comment]
@dubdabasoduba yeah, `styles.xml`

**Comment by FikriMilano** [comment]
@f-odhiambo 
Analyzed these, whether the requirements needs to be implemented on FHIRCore or SDK or even both.

SS 1
1. Header properties
   - [ ] Should have a close button instead of back button when the form is scrolled **_[FHIRCORE]_**
   - [ ] The header should have a dropdown shadow **_[FHIRCORE]_**

2. Bottom button and container
   - [ ] Button container should have drop shadow if there’s more content down the list **_[FHIRCORE]_**
   - [ ] The buttons have round edges in design and color of buttons is different - see action colors here **_[FHIRCORE]_**

3. Questions 
   - [ ] The questions font weights and size is different from designs. We use big font in these forms for **_[FHIRCORE]_**
   - [ ] For required questions, we shouldn’t use “*”, instead if a question is optional that’s when we will indicate the “(optional)” badge/text **_[SDK]_**

4. Spacing 
   - [ ] The spacing between the question and the body has been updated from 24px to 16px.  **_[SDK]_** **_[FHIRCORE]_**
   - [ ] And the space between questions should be 32px  **_[SDK]_** **_[FHIRCORE]_**

5. Select Options/Answers
   - The design properties of these components are different from the original designs.
   - [ ] Text is smaller **_[FHIRCORE]_**
   - [ ] Padding not the same as in design **_[SDK]_** **_[FHIRCORE]_**
   - [ ] Icons used are different **_[FHIRCORE]_**
   - [ ] Colors used are different **_[FHIRCORE]_**
   - [ ] Spacing between items in the components different from design **_[SDK]_** **_[FHIRCORE]_**

SS 2
1. Header navigation 
   - [ ] Use close button and not back button **_[FHIRCORE]_**

2. Spacing  
   - [ ] Space between input label and the text field is more than in design **_[SDK]_** **_[FHIRCORE]_**

3. Input field label 
   - [ ] Text field labels are bolder in the screenshot, design proposed regular **_[FHIRCORE]_**
   - [ ] The optional fields should have tag “(optional)” and required fields should have none **_[SDK]_**

4. Input field
   - [ ] Input field borders have different color from design **_[FHIRCORE]_**

5. Save button 
   - [ ] Input field borders have different color from design **_[FHIRCORE]_**
   - [ ] And the corner radius **_[FHIRCORE]_**

SS 3
1. Header Navigation 
   - [ ] Use close button and not back button **_[FHIRCORE]_**
   - [ ] Top header navigation should have drop shadow on scroll **_[FHIRCORE]_**

2. Save button 
   - [ ] Input field borders have different color from design **_[FHIRCORE]_**
   - [ ] And the corner radius **_[FHIRCORE]_**
   - [ ] The container should have dropdown shadow in the upward direction when there’s content below it, and just a border line at the top if no content below it i.e at end of the list **_[FHIRCORE]_**

**Comment by shelaghm** [comment]
@FikriMilano Thanks for looping me in. I see there's a couple of SDK padding issues, I'm hopeful that we can make these changes in the SDK. Can you file an issue on the SDK github with what the ideal padding would be and tag me in it? Thanks!

**Comment by FikriMilano** [comment]
@shelaghm thanks for commenting!

I will surely file an issue on the SDK and tag you.

**Comment by FikriMilano** [comment]
@f-odhiambo @dubdabasoduba is this still low priority? or do we need to get this merged soon?

**Comment by FikriMilano** [comment]
Blocked by https://github.com/google/android-fhir/pull/1852

**Comment by f-odhiambo** [comment]
@FikriMilano This is now unblocked

**Comment by FikriMilano** [comment]
Good reminder @f-odhiambo

**Comment by f-odhiambo** [comment]
@HenryRae to review and share an update on the widget styling for this ticket

**Comment by HenryRae** [comment]
@FikriMilano @Rkareko 

1. Paginated forms can have text buttons "back", "next" and "save/done"(any copy defined) at the last paginated form. See example [here](https://www.figma.com/file/gDW2MxoHNI77hv1O49xgAk/FHIR-(OpenSRP)-Apps-~-Design-System?type=design&node-id=20617%3A247952&mode=dev&t=AbdQFxoRYydrBBeb-1)

2. Non-paginated forms should have use a primary rectangular block button with background color. See example [here](https://www.figma.com/file/gDW2MxoHNI77hv1O49xgAk/FHIR-(OpenSRP)-Apps-~-Design-System?type=design&node-id=20600%3A241314&mode=dev&t=AbdQFxoRYydrBBeb-1)

**Summary:** _Use text buttons on paginated forms, and rectangular buttons with background color on non-paginated forms_
